### PR TITLE
Split network topology code into separate file

### PR DESCRIPTION
**Pull request by gcongiu**
## Pull Request Description

This PR groups network topology functions currently located in `mpir_hw_topo.c` into a separate file and renames functions and data structures accordingly. The goal of this PR is only to separate the relevant network topology code from old topology implementation as this was replaced by #4127 for the node topology part. Future PR will rework the network topology code starting from here.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by gcongiu** [comment]
test:jenkins/ch3/tcp

**Comment by gcongiu** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by gcongiu** [comment]
@hzhou this seems fine if you want to give it another pass.

File: src/mpi/init/finalize.c [hunk 1]
```diff
@@ -159,6 +159,7 @@ int MPI_Finalize(void)
 
     MPII_hw_topo_finalize();    /* to be eventually replaced by MPII_hwtopo_finalize() */
```
Comment (hzhou) [inline src/mpi/init/finalize.c]: Which commit eventually removes this line?

File: src/mpi/init/initthread.c [hunk 2]
```diff
@@ -93,6 +93,7 @@ int MPIR_Init_thread(int *argc, char ***argv, int required, int *provided)
 
     MPII_hw_topo_init();        /* to be eventually replaced by MPII_hwtopo_init() */
```
Comment (hzhou) [inline src/mpi/init/initthread.c]: Same question.

File: src/mpi/init/finalize.c [hunk 3]
```diff
@@ -157,7 +157,6 @@ int MPI_Finalize(void)
     /* Call the low-priority (post Finalize) callbacks */
     MPIR_Call_finalize_callbacks(0, MPIR_FINALIZE_CALLBACK_PRIO - 1);
 
-    MPII_hw_topo_finalize();    /* to be eventually replaced by MPII_hwtopo_finalize() */
```
Comment (hzhou) [inline src/mpi/init/finalize.c]: Oh, it is here. I guess it doesn't fit the commit message. Could you move to the next commit?

File: src/mpi/init/initthread.c [hunk 4]
```diff
@@ -91,7 +91,6 @@ int MPIR_Init_thread(int *argc, char ***argv, int required, int *provided)
     mpi_errno = MPII_init_global(&required);
     MPIR_ERR_CHECK(mpi_errno);  /* out-of-mem */
 
-    MPII_hw_topo_init();        /* to be eventually replaced by MPII_hwtopo_init() */
```
Comment (hzhou) [inline src/mpi/init/initthread.c]: same.

File: src/mpi/init/finalize.c (hunk 1 above)
Comment (gcongiu) [reply src/mpi/init/finalize.c]: I think I have removed these in the commit before the last. I should move these to the last commit.

**Comment by gcongiu** [comment]
test:jenkins/ch3/tcp

**Comment by gcongiu** [comment]
@hzhou fyi I added a last commit to remove `init_topo.c` that somehow was still there although it was not compiled. It still had the `MPIR_CVAR_NETLOC_NODE_FILE` definition in it, which I also moved to `mpir_nettopo.c`. I am rerunning CH3 tests as precaution.

**Pull request merged**